versions/src/main/java/cc/polyfrost/oneconfig/utils/commands/arguments/PlayerArgumentParser.java

```diff
@@ -1,33 +1,71 @@
-//#if MC<=11202
 package cc.polyfrost.oneconfig.utils.commands.arguments;
 
+import cc.polyfrost.oneconfig.utils.NetworkUtils;
 import com.google.common.collect.Lists;
+import com.google.gson.JsonObject;
+import com.mojang.authlib.GameProfile;
 import net.minecraft.client.Minecraft;
-import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.client.network.NetworkPlayerInfo;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import java.lang.reflect.Parameter;
+import java.util.HashMap;
 import java.util.List;
+import java.util.UUID;
 import java.util.stream.Collectors;
 
-public class PlayerArgumentParser extends ArgumentParser<EntityPlayer> {
+/**
+ * The player argument parser. Returns a {@link GameProfile}.
+ */
+public class PlayerArgumentParser extends ArgumentParser<GameProfile> {
+    private static final HashMap<String, UUID> uuidCache = new HashMap<>();
     @Nullable
     @Override
-    public EntityPlayer parse(@NotNull String arg) throws Exception {
-        return getMatchingPlayers(arg).stream().findFirst().orElseThrow(() -> new Exception("No player found"));
+    public GameProfile parse(@NotNull String arg) {
+        List<GameProfile> matchingPlayers = getMatchingPlayers(arg, false);
+        for (GameProfile profile : matchingPlayers) {
+            return profile;
+        }
+        return new GameProfile(getUUID(arg), arg);
     }
 
-    public static List<EntityPlayer> getMatchingPlayers(String arg) {
+    // checks mojang api for player uuid from name
```

Comment:
I'll be real why do we have this it is just a waste of processing time and memory they can implement this themselves if they want

Reply:
so they can put nonexistant players if they want (for stuff like hytils gexp checker)